sql/core/src/main/scala/org/apache/spark/sql/execution/GenerateExec.scala

```diff
@@ -85,11 +84,19 @@ case class GenerateExec(
     val numOutputRows = longMetric("numOutputRows")
     child.execute().mapPartitionsWithIndexInternal { (index, iter) =>
       val generatorNullRow = new GenericInternalRow(generator.elementSchema.length)
-      val rows = if (join) {
+      val rows = if (requiredChildOutput.nonEmpty) {
+
+        val pruneChildForResult: InternalRow => InternalRow =
+          if ((child.outputSet -- requiredChildOutput).isEmpty) {
```

Comment:
just `child.output == requiredChildOutput`?

Reply:
wouldn't it always return false? or should I use `child.output == AttributeSet(requiredChildOutput)`